lib/system_themes_manager.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class SystemThemesManager
+  def self.sync!
+    Theme::CORE_THEMES.keys.each { |theme_name| sync_theme!(theme_name) }
+  end
+
+  def self.sync_theme!(theme_name)
+    theme_id = Theme::CORE_THEMES[theme_name]
+    return unless theme_id
+
+    theme_dir = "#{Rails.root}/themes/#{theme_name}"
+    return if !Dir.exist?(theme_dir)
```

Comment:
This line does not feel right to me. Even if the directory exists, the content of the directory may be different.